src/Domain/Creators/TimeDependence/Composition.tpp

```diff
@@ -0,0 +1,146 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Domain/Creators/TimeDependence/Composition.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+namespace creators {
+namespace time_dependence {
+namespace detail {
+template <typename... MapsSoFar>
+auto combine_coord_maps(
+    CoordinateMap<Frame::Grid, Frame::Inertial, MapsSoFar...>
+        coord_map_so_far) noexcept {
+  return coord_map_so_far;
+}
+
+template <typename... MapsSoFar, typename... NextMaps, typename... RestMaps>
+auto combine_coord_maps(
+    CoordinateMap<Frame::Grid, Frame::Inertial, MapsSoFar...> coord_map_so_far,
+    CoordinateMap<Frame::Grid, Frame::Inertial, NextMaps...> next_map,
+    RestMaps... rest_maps) noexcept {
+  return combine_coord_maps(domain::push_back(coord_map_so_far, next_map),
+                            rest_maps...);
+}
+
+template <typename... Ts>
+std::unordered_map<std::string,
+                   std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+combine_functions_of_time(const std::vector<std::string>& time_dep_names,
+                          Ts... functions_of_time_pack) noexcept {
+  static_assert(sizeof...(Ts) > 0,
+                "Must have at least one set of function of times to combine. "
+                "We must have one "
+                "entry per TimeDependence being composed.");
+
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+  size_t counter_into_dep_names = 0;
+
+  const auto insert_into_map = [&counter_into_dep_names, &functions_of_time,
+                                &time_dep_names](
+                                   std::unordered_map<
+                                       std::string,
+                                       std::unique_ptr<domain::FunctionsOfTime::
+                                                           FunctionOfTime>>&&
+                                       functions_of_time_single_map) noexcept {
+    for (auto& name_and_func : functions_of_time_single_map) {
+      if (UNLIKELY(functions_of_time.count(name_and_func.first) != 0)) {
```

Comment:
Sorry, I'm sure I've asked this before, but can you remind me why we are using a map rather than some form of type-indexed object (like a tagged tuple) for the functions of time storage?

Reply:
Because we don't know what functions of time exist until runtime. The reason for this is that coordinating the control system and the domain is really non-trivial and we couldn't figure out a good way of doing it so the decision was made to first get moving mesh and control systems done and working, then consider revisiting if something comes along at that point that could work. Does that clear things up?

Reply:
Yeah, that makes sense. Thanks!